invenio_search/ext.py

```diff
@@ -244,49 +240,132 @@ def active_aliases(self):
         else:
             return {k: v for k, v in self.aliases.items()
                     if k in whitelisted_aliases}
 
+    def _get_indices(self, tree_or_filename):
+        for name, value in tree_or_filename.items():
+            if isinstance(value, dict):
+                for result in self._get_indices(value):
+                    yield result
+            else:
+                yield name
+
+    def create_index(self, index, mapping_path=None, prefix=None, suffix=None,
+                     create_write_alias=True, ignore=None, dry_run=False):
+        """Create index with a write alias."""
+        mapping_path = mapping_path or self.mappings[index]
+
+        final_alias = None
+        final_index = None
+        index_result = None, None
+        alias_result = None, None
+        # To prevent index init --force from creating a suffixed
+        # index if the current instance is running without suffixes
+        # make sure there is no index with the same name as the
+        # alias name (i.e. the index name without the suffix).
+        with open(mapping_path, 'r') as body:
+            final_index = build_index_name(
+                index, prefix=prefix, suffix=suffix, app=self.app)
+            if create_write_alias:
+                final_alias = build_alias_name(
+                    index, prefix=prefix, app=self.app)
+            index_result = (
+                final_index,
+                self.client.indices.create(
+                    index=final_index,
+                    body=json.load(body),
+                    ignore=ignore,
+                ) if not dry_run else None
+            )
+            if create_write_alias:
+                alias_result = (
+                    final_alias,
+                    self.client.indices.put_alias(
+                        index=final_index,
+                        name=final_alias,
+                        ignore=ignore,
+                    ) if not dry_run else None
+                )
+        return index_result, alias_result
+
     def create(self, ignore=None):
         """Yield tuple with created index name and responses from a client."""
         ignore = ignore or []
-
-        def _create(tree_or_filename, alias=None):
-            """Create indices and aliases by walking DFS."""
-            # Iterate over aliases:
+        new_indices = {}
+        actions = []
+
+        def check_exists(name):
+            if current_search_client.indices.exists(name):
+                raise IndexAlreadyExistsError(
+                    'index with name "{}" already exists'.format(name))
+            if current_search_client.indices.exists_alias(name, name='*'):
+                raise IndexAlreadyExistsError(
+                    'alias with name "{}" already exists'.format(name))
+
+        def _build(tree_or_filename, alias=None):
+            """Create indices and aliases by walking DFS.
+
+            Yields the result from create_index() and boolean is_index.
```

Comment:
This can be either removed (since nothing is yielded/returned). Maybe it's worth stating that it populates the `actions` list with dicts that have a specific structure.